lib/libutee/tee_api_objects.c

```diff
@@ -237,21 +305,64 @@ void TEE_InitValueAttribute(TEE_Attribute *attr, uint32_t attributeID,
 	attr->content.value.b = b;
 }
 
+/*
+ * Use of this function is deprecated
+ * new code SHOULD use the TEE_CopyObjectAttributes1 function instead
+ * These functions will be removed at some future major revision of
+ * this specification
+ */
 void TEE_CopyObjectAttributes(TEE_ObjectHandle destObject,
 			      TEE_ObjectHandle srcObject)
 {
 	TEE_Result res;
+	TEE_ObjectInfo src_info;
+
+	res = utee_cryp_obj_get_info((uint32_t)srcObject, &src_info);
+	if (src_info.objectType == TEE_TYPE_CORRUPTED_OBJECT)
+		return;
+
+	res = TEE_CopyObjectAttributes1(destObject, srcObject);
+	if (res != TEE_SUCCESS)
+		TEE_Panic(0);
+}
+
+TEE_Result TEE_CopyObjectAttributes1(TEE_ObjectHandle destObject,
+			      TEE_ObjectHandle srcObject)
+{
+	TEE_Result res;
 	TEE_ObjectInfo dst_info;
 	TEE_ObjectInfo src_info;
 
 	res = utee_cryp_obj_get_info((uint32_t)destObject, &dst_info);
-	if (res != TEE_SUCCESS)
+	if (res != TEE_SUCCESS && res != TEE_ERROR_CORRUPT_OBJECT &&
+	    res != TEE_ERROR_STORAGE_NOT_AVAILABLE)
 		TEE_Panic(0);
 
+	if (res == TEE_ERROR_CORRUPT_OBJECT) {
```

Comment:
If this test and action was done before the test for panic at line 237-238 above, it would simplify the test for panic.